src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -4158,20 +4159,23 @@ private static BoundExpression SkipReferenceConversions(BoundExpression possibly
             BoundExpression leftOperand = node.LeftOperand;
             BoundExpression rightOperand = node.RightOperand;
 
-            TypeWithState leftResult = VisitRvalueWithState(leftOperand);
-            TypeWithState rightResult;
-
+            // PROTOTYPE(improved-definite-assignment):
+            // Definite assignment has another special case for when the LHS is a non-null constant (a literal string for example).
+            // Should we replicate that here?
```

Comment:
I think so. It reduces false positive warnings by skipping unreachable code.

Reply:
Done

Reply:
https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4GIYFcA2OLA4CmABETAcQLABQtAAgEwCMttA9OyQE5QDmACwAuJAM5QEpUQID2uBCWCkARFhjciEAMYDKRZbRnAAVkS1CA/CQAeAbloB9EgF4SygILKSFqwBJlAN7WAHQAKjIAykK8MHwAFACUAL7K9nQ0nDz8wmISUrLyiqQAblDiQkQKaloyMAhQQlC1EHgAnoYmZpYkrWlOrh7KFsEAogCOWC2ica0uJAAMCWmtYZHRsPFLbDT0zACcccqAPBuAIPvKS0A===

Looks like both analyses already do I had in mind for both constructs. Will add some tests to confirm.